packages/sonarwhal/src/lib/cli/analyze.ts

```diff
@@ -204,15 +208,15 @@ export const analyze = async (actions: CLIOptions): Promise<boolean> => {
         });
     };
 
-    const print = (reports: Array<Problem>, target: string) => {
+    const print = async (reports: Array<Problem>, target: string): Promise<void> => {
         if (hasError(reports)) {
             endSpinner('fail');
         } else {
             endSpinner('succeed');
         }
 
-        sonarwhal.formatters.forEach((formatter) => {
-            formatter.format(reports, target);
+        await each(sonarwhal.formatters, async (formatter) => {
```

Comment:
why do you use the package `async` instead of use a `for … of..`?

Reply:
We should start using `async` in all places instead of `for … of` for legibility purposes and this was a good moment to start.

Reply:
`for … of` is legible :P

Reply:
Let's replace all the `map`, `reduce`, `forEach`, etc. for regular `for` loops then 😛

Reply:
It is not the same example, you don't need to add any extra package to do that.

Reply:
We are using `lodash` in parts of the code 🙄